light-client-verifier/src/verifier.rs

```diff
@@ -146,29 +158,51 @@ where
         Verdict::Success
     }
 
-    /// Validate an `UntrustedBlockState`, based on the given `TrustedBlockState`, `Options` and
-    /// current time.
+    /// Validate an `UntrustedBlockState` coming from a client update,
+    /// based on the given `TrustedBlockState`, `Options` and current time.
     pub fn validate_against_trusted(
         &self,
         untrusted: &UntrustedBlockState<'_>,
         trusted: &TrustedBlockState<'_>,
         options: &Options,
         now: Time,
     ) -> Verdict {
-        // Ensure the latest trusted header hasn't expired
-        verdict!(self.predicates.is_within_trust_period(
-            trusted.header_time,
-            options.trusting_period,
-            now,
-        ));
-
         // Ensure the header isn't from a future time
         verdict!(self.predicates.is_header_from_past(
             untrusted.signed_header.header.time,
             options.clock_drift,
             now,
         ));
 
```

Comment:
I was wondering why this check is in a function called "validate_against_trusted" as it doesn't need the trusted state.

Reply:
Maybe that's cleaner indeed!

Reply:
9e0a4c851cbd9894f12b5806a57e0eb8ee757ef9